base/applications/rapps/include/rosui.h

```diff
@@ -374,173 +396,150 @@ class CUiPanel :
         m_Height = CUiMeasure::FitParent();
     }
 
-    virtual ~CUiPanel()
-    {
-    }
+    virtual ~CUiPanel() {}
 
-    virtual CUiBox * AsBox() { return this; }
+    // virtual CUiBox* AsBox() { return this; }
 
     virtual VOID ComputeMinimalSize(SIZE* size)
     {
         for (INT i = 0; i < m_Children.GetCount(); i++)
         {
-            CUiBox * box = m_Children.Get(i)->AsBox();
-            if (box)
-            {
-                box->ComputeMinimalSize(size);
-            }
+            m_Children.Get(i)->ComputeMinimalSize(size);
         }
-    };
+    }
 
     virtual VOID ComputeContentBounds(RECT* rect)
     {
         for (INT i = 0; i < m_Children.GetCount(); i++)
         {
-            CUiBox * box = m_Children.Get(i)->AsBox();
-            if (box)
-            {
-                box->ComputeContentBounds(rect);
-            }
+            m_Children.Get(i)->ComputeContentBounds(rect);
         }
-    };
+    }
 
-    virtual DWORD_PTR CountSizableChildren()
+    virtual INT CountSizableChildren()
     {
         INT count = 0;
         for (INT i = 0; i < m_Children.GetCount(); i++)
         {
-            CUiBox * box = m_Children.Get(i)->AsBox();
-            if (box)
-            {
-                count += box->CountSizableChildren();
-            }
+            count += m_Children.Get(i)->CountSizableChildren();
         }
         return count;
     }
 
     virtual HDWP OnParentSize(RECT parentRect, HDWP hDwp)
     {
-        RECT rect = {0};
-
         SIZE content = {0};
         ComputeMinimalSize(&content);
 
-        INT preferredWidth = m_Width.ComputeMeasure(parentRect.right - parentRect.left, content.cx);
-        INT preferredHeight = m_Height.ComputeMeasure(parentRect.bottom - parentRect.top, content.cy);
-
-        rect.right = preferredWidth;
-        rect.bottom = preferredHeight;
-
+        RECT rect = {0};
+        rect.right = m_Width.ComputeMeasure(parentRect.right - parentRect.left, content.cx);
+        rect.bottom = m_Height.ComputeMeasure(parentRect.bottom - parentRect.top, content.cy);
         ComputeRect(parentRect, rect, &rect);
 
         for (INT i = 0; i < m_Children.GetCount(); i++)
         {
-            CUiBox * box = m_Children.Get(i)->AsBox();
-            if (box)
-            {
-                hDwp = box->OnParentSize(rect, hDwp);
-            }
+            hDwp = m_Children.Get(i)->OnParentSize(rect, hDwp);
         }
 
         return hDwp;
     }
 };
 
-template<class T = CWindow>
+
+//
+// Specifications for implementing window layout, and splitter
+//
+
+template<class TBase = CWindow>
 class CUiWindow :
     public CUiPrimitive,
-    public CUiBox,
-    public T
+    // public CUiBox,
+    public TBase
 {
 public:
-    virtual CUiBox * AsBox() { return this; }
-
-    HWND GetWindow() { return T::m_hWnd; }
```

Comment:
This is the function that is not necessary since we have (by default) `m_hWnd` that is public (by ATL's `CWindow` or derived classes), and the implementation of `CUiWindow` makes it implicit, by the usage of this member as well as other window functions, that we will only accept template specialization with other classes that "look like" ATL's `CWindow`.